app/code/Magento/Sales/Model/Order/Shipment/Track.php

```diff
@@ -415,5 +444,17 @@ public function setExtensionAttributes(\Magento\Sales\Api\Data\ShipmentTrackExte
         return $this->_setExtensionAttributes($extensionAttributes);
     }
 
+    /**
+     * Get Track Url for Track Item and return only if valid, absolute URI
+     *
+     * @return string|bool
+     */
+    public function getValidTrackUrl()
```

Comment:
Looks like this method's only usage occurs in the block. Therefore it may be moved to the block or even a separate service.